Update der Seite "Die Docker Devilbox" (Änderungen aus Devilbox V3 beta)

Ernestopheles opened the pull request:
Weiter geht es mit den weiteren neuen Funktionalitäten, coming soon

Ernestopheles committed:
1. Teil Update der Doku

Weiter geht es mit den weiteren neuen Funktionalitäten, coming soon

Ernestopheles committed:
Update devilbox.de.md

Ernestopheles commented:
Jetzt ist das  Thema von meiner Seite aus vollständig behandelt.

Ernestopheles committed:
Update devilbox.de.md

Kopierfehler entfernt

fkaminski commented:
Grundsätzlich sollten wir mit den Änderungen warten bis die erste, stabile devilbox 3er Version veröffentlicht ist. Entsprechend kann man daher auf jeweilige »Beta« Hinweise verzichten. Ein einmaliger Hinweis ala:
"Die folgenden Angaben beziehen sich auf die devilbox ab der Version 3.0" sollte dann ausreichen.

Ernestopheles commented:
Cytopia von devilbox hat um Unterstützung gebeten für den Test des Beta Releases. Ich denke von daher wäre jetzt schon der richtige Zeitpunkt für die Veröffentlichung.

Das Aufräumen der Container wurd in der Devilbox Doku strikt empfohlen, es geht nichts dabei verloren.

Ernst
Am 15. Jan. 2023, 13:21 +0100 schrieb Franko ***@***.***>:
> @fkaminski commented on this pull request.
> In docs/manual/guides/local-installation/devilbox.de.md:
> > +Die Devilbox sollte durch den Stop aller Container und anschliessendem Löschen aller Container beendet werden.
>
>
>
> ```bash
>
> docker-compose stop
>
> +docker-compose rm -f
>
> ```
>
> Die Devilbox kann durch den Stop aller Container beendet werden. Die Inhalte der Container werden dabei nicht gelöscht.
> docker-compose stop
> Die Info »docker-compose rm -f« ist kritisch, da hierbei alles entfernt wird. Würde ich nicht hier so aufführen. Oder als separaten Eintrag behandeln.
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you authored the thread.Message ID: ***@***.***>

fkaminski commented:
?? »docker-compose rm«:
https://docs.docker.com/engine/reference/commandline/compose_rm/

Für Contao Anwender eher verwirrend. Oder man erstellt diesbezüglich ein neues Kapitel zwecks permanenter Löschung von container/images/binds/volumes etc. - Aber ist wohl im Kontext eher zu spezifisch. Dafür kann man die docker Doku nutzen.

fkaminski commented:
> Cytopia von devilbox hat um Unterstützung gebeten für den Test des Beta Releases. Ich denke von daher wäre jetzt schon der richtige Zeitpunkt für die Veröffentlichung.

Schon. Aber sowas gehört dann ins devilbox repo. Hier geht es um die Doku für Contao Anwender.

fkaminski commented:
Ich empfinde die neue Aufteilung der Chapter bzw. die Aufnahme der neuen devilbox Funktionalitäten für Contao Anwender eher verwirrend. Ausgehend von der Prämisse, dass die meisten Contao Anwender eher "nur" ein Projekt lokal halten, würde die bisherige Chapter-Aufteilung genügen und einfach nachvollziehbar bleiben. 

Entsprechend bedarf es, sofern erforderlich, lediglich Updates der bestehenden Kapitel hinsichtlich:
- Contao document-root Eintrag (web/public)
- Hinweise zur Änderung der jeweiligen PHP ini Einträge per Version  ...
- ... Angabe zum PHP [memory-limit](https://github.com/cytopia/devilbox/issues/469) (composer Installation via CLI) ist ausreichend (bzw. eh nur für PHP composer Nutzung auf der CLI relevant )
- Hinweise auf .loc Domain Nutzung bzw. AUTO-DNS Feature der devilbox
- Definition/Nutzung einer »Default« PHP Version

Entsprechende Angaben (in der korrekten Reihenfolge) sind bereits vorhanden und sollten, wie erwähnt, lediglich für die möglichen Änderungen des devilbox 3er release ergänzt werden. Eine einfache, nachvollziehbare StepByStep Lösung für Contao Anwender zwecks lokaler Installation bliebe somit gegeben.

Die neuen Features der devilbox sind wichtig u. hilfreich (u.a. hat @Ernestopheles auch die xDebug Config aufgenommen). 
Allerdings würde ich diese gänzlich separat, unterhalb der bestehende Chapter (Easy Step) aufführen/sehen - 
also für »advanced using« - als eigene (unabhängig von den obigen easy one project steps) Kapitel:

- Mehrere Contao Projekte mit versch. PHP Versionen konfigurieren
- xDebug & die devilbox

Ernestopheles commented:
https://devilbox.readthedocs.io/en/latest/getting-started/start-the-devilbox.html#stop-and-restart
Am 15. Jan. 2023, 13:40 +0100 schrieb Franko ***@***.***>:
> ?? »docker-compose rm«:
> https://docs.docker.com/engine/reference/commandline/compose_rm/
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you authored the thread.Message ID: ***@***.***>

fkaminski commented:
> https://devilbox.readthedocs.io/en/latest/getting-started/start-the-devilbox.html#stop-and-restart 

Ich glaube »docker-compose rm -f« wäre nur notwendig, sofern neuere docker images o. ein [Update the Devilbox](https://devilbox.readthedocs.io/en/latest/maintenance/update-the-devilbox.html) etc. gezogen werden. Etwaige Einstellungen innerhalb der bestehenden container (z.B. ein php composer update etc.) gehen doch dabei verloren. Oder man sollte zur Sicherheit hierbei hervorheben, das man zuvor entsprechende Backups (z.B. Datenbanken) erstellen sollte.
Für den "normalen" Betrieb reicht ein Start/Stop der Container aus.

Ernestopheles commented:
Ich arbeite jetzt schon länger mit Devilbox und bisher ging nichts verloren. PHP Updates macht Devilbox im Hintergrund und Contao Updates bleiben selbstverständlich erhalten

Ernst
Am 16. Jan. 2023, 08:41 +0100 schrieb Franko ***@***.***>:
> > https://devilbox.readthedocs.io/en/latest/getting-started/start-the-devilbox.html#stop-and-restart
> Ich glaube »docker-compose rm -f« wäre nur notwendig, sofern neuere docker images o. devlibox release etc. gezogen werden. Etwaige Einstellungen innerhalb der bestehenden container (z.B. ein php composer update etc.) gehen doch dabei verloren. Oder man sollte zur Sicherheit hierbei hervorheben, das man zuvor entsprechende Backups (z.B. Datenbanken) erstellen sollte.
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.Message ID: ***@***.***>

fkaminski commented:
Kann sein, das ich es mit »[docker rm](https://docs.docker.com/engine/reference/commandline/rm/)« verwechselt habe.

Ernestopheles committed:
Update devilbox.de.md

Added a link to DNS Server router configuration

Ernestopheles commented:
> Kann sein, das ich es mit »[docker rm](https://docs.docker.com/engine/reference/commandline/rm/)« verwechselt habe.

Devilbox nutzt das volume mounting von docker. Die Datenbank wird vom Host gemounted und steht damit auch nach dem Löschen der container zur Verfügung (vgl. https://docs.docker.com/get-started/05_persisting_data/)

Das gleiche gilt wohl auch für das Projektverzeichnis.

Ernestopheles commented:
> Die neuen Features der devilbox sind wichtig u. hilfreich (u.a. hat @Ernestopheles auch die xDebug Config aufgenommen). Allerdings würde ich diese gänzlich separat, unterhalb der bestehende Chapter (Easy Step) aufführen/sehen - also für »advanced using« - als eigene (unabhängig von den obigen easy one project steps) Kapitel:
> 
> * Mehrere Contao Projekte mit versch. PHP Versionen konfigurieren
> * xDebug & die devilbox

Verstehe leider nicht, was du mit easy step und easy one project steps meinst. 

Ansonsten überlasse ich dir auch gerne das Feld, es ist gar nicht so einfach sich zu einigen und du hast die meiste Erfahrung

fkaminski commented:
> Verstehe leider nicht, was du mit easy step und easy one project steps meinst.

Ziel ist die einfache, lokale Contao Installation via devilbox zu zeigen. Hierbei sollten ausschließlich die dazu notwendigen, minimalen Angaben zwecks einfacher Vorgehensweise aufgezeigt werden ( Idealerweise durch Änderung von ein paar .env Variablen u. starten der devilbox ):

**Beispiel I:** 
Daher wurde z. B. der Eintrag »**HTTPD_SERVER=apache-2.4**« favorisiert. Zwar könnte man auch nginx verwenden. Dies wäre aber u.U. mit weiteren Konfigurationen im Kontext einer Contao Installation verbunden, was möglicherweise Contao Anwender abschreckt ( Wer dies möchte kann hierzu die devilbox Doku bemühen ). 

**Beispiel II:**
Die weitere PHP Konfiguration ist "normalerweise" nicht notwendig ( zumindest nicht zusammen mit einer Contao Manager Installation via Resolver ). Notwendig wird dies erst bei Nutzung von PHP Composer via CLI. 

Ich würde daher die bisherigen Angaben im Kapitel "[Installation über die Kommandozeile](https://docs.contao.org/manual/de/anleitungen/lokale-installation/devilbox/#installation-über-die-kommandozeile)" so belassen ( PHP memory limit ) bzw. mit deinen Angaben aus "**php Konfiguration**" ergänzen. Ansonsten diesen Bereich des PR's dann aus dem Abschnitt "**Contao-Installation vorbereiten**" entfernen.

**Beispiel III:**
Deine Angaben im Kapitel "**Contao-Installation vorbereiten**" des PR's bez. der neuen Domains ( .dvl.to ) sind ausreichend: 
_... Ab devilbox Version 3 ..._

Die weiteren Notice Angaben 
_Falls bei dir URLs mit dem Suffix dvl.to nicht aufgelöst werden können, dann kann das an den DNS Server Einstellungen in deinem Netzwerk liegen ..._

würde ich, gemäß obiger Prämisse, entfernen. Für etwaige Fallstricke wäre die devilbox zu konsultieren bzw. zu referenzieren 
( ähnlich wie es bisher bez. der "Auto-DNS" Funktionalität erfolgt ).

**Beispiel IV:**
Manche Dinge die für dich selbstverständlich sind, stellen möglicherweise andere vor Probleme. 
Allgemeiner ( u.a. f. Windows User ) wäre z. B. im Kapitel "**Mehrere PHP Versionen parallel betreiben**" etwas wie:

_Die Grundeinstellung nimmst du im devilbox Verzeichnis vor. Im Verzeichnis »compose« findest du die Datei »docker-compose.override.yml-php-multi.yml«. Kopiere die Datei in das devilbox Hautpverzeichnis und ändere den Dateinamen nach »docker-compose.override.yml«. Damit werden die ...
Anschließend kannst du je Projekt, abweichend von der in der ».env« eingestellten PHP-Version, gezielt ..._

In der Art / JM2C
Letztlich entscheidet das Doku-Team

netzarbeiter commented:
Vielen Dank für den Pull Request, Ernst.

Wie Franko schon geschrieben hat, das Ziel des Eintrages sollte folgendes sein:

Ich möchte Contao lokal installieren und weiss nicht wie ich bei Docker Devilbox vorgehen muss. In der Anleitung werde ich für eine Grundinstallation an die Hand genommen. Für tiefergreifende Informationen einfach auf die Doku von Devilbox verweisen.

Es gilt für unsere Anleitungen immer, so kurz wie möglich und so lange wie nötig.

Kannst du die Anleitung bitte auf die dahingehend wesentlichen Dinge reduzieren.

Ernestopheles committed:
Update devilbox.de.md

Ich muss jetzt weg, aber ich glaube ich habe eh alles wunschgemäß geändert, hoffe ich ;-)) Ansonsten sag Bescheid

Ernestopheles committed:
Update devilbox.de.md

done

fkaminski commented:
Die Änderungen lesen sich gut. 
Evtl. würde ich bei den Versionsangaben/Voreinstellungen jeweils Contao 4.13 LTS u. PHP 8.2 benutzen?

**Mehrere PHP Versionen parallel betreiben**
Hier würde ich auch konkrete Angaben/Beispiele bevorzugen (anstatt xy):
_conf:phpfpm:tcp:php82:9000_

Zusätzlich wäre es hier hilfreich die Tabelle "**PHP hostnames and IP addresses**" aufzunehmen aus:
https://github.com/cytopia/devilbox/releases/tag/v3.0.0-beta-0.1

**PHP Xdebug**
Hier würde ich die Beispiele »launch.json/php.ini« via Tabs/Register darstellen:
https://learn.netlify.app/en/shortcodes/tabs/

Ernestopheles committed:
Update devilbox.de.md

Danke, Franko, für deine Tipps. Hoffe, es ist so ok. Leider ist die Vorschau bei einigen Elementen wie z.B. tabs nicht so wie es später aussieht. 

Die angesprochene Tabelle ist doch speziell für Nginx, würde ich daher nicht so sehr empfehlen, aufzunehmen.

fkaminski commented:
Könntest du »**Mehrere PHP Versionen parallel betreiben**« bitte noch ergänzen ala:

_...
conf:phpfpm:tcp:php74:9000
Die aktuell verfügbaren Hostnamen kannst du dieser Liste entnehmen:_

Und dann unteren [Screen](https://github.com/cytopia/devilbox/releases/tag/v3.0.0-beta-0.1) als Markdown table einfügen? 

![image](https://user-images.githubusercontent.com/7679799/212978110-b15c52f4-fb07-4bda-86c3-ff4a989766d0.png)

Ernestopheles commented:
Kann ich machen, aber ich halte es für unnötig und verwirrend. 
Der Hostname ergibt sich doch sofort aus der gewünschten php Version...


Am Dienstag, dem 17.01.2023 um 10:20 -0800 schrieb Franko:
> Könntest du »Mehrere PHP Versionen parallel betreiben« bitte noch
> ergänzen ala:
> ...
> conf:phpfpm:tcp:php74:9000
> Die aktuell verfügbaren Hostnamen kannst du dieser Liste entnehmen:
> Und dann unteren Screen als Markdown table einfügen aus
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>

fkaminski commented:
> Der Hostname ergibt sich doch sofort aus der gewünschten php Version...

Das ist sicherlich richtig! Zwecks Übersicht/Verständnis aber evtl. dennoch hilfreich. 
Nicht zuletzt wegen der kleinen aber feinen Anmerkung:
_Note: Use IP addresses instead of hostnames for Nginx_ 
:-)

Oder wir verlinken dann später hierzu in einer Notiz auf die devilbox Doku (sofern dann verfügbar).
Letztlich wird aber sicherlich @netzarbeiter nochmals gegenlesen und final entscheiden.

Ernestopheles commented:
Ah ja, wegen Nginx wäre es sinnig. Aber wollten wir solche Details
nicht fortlassen weil das Spezialfälle sind?

Am Dienstag, dem 17.01.2023 um 10:51 -0800 schrieb Franko:
> > Der Hostname ergibt sich doch sofort aus der gewünschten php
> > Version...
> Das ist sicherlich richtig! Zwecks Übersicht/Verständnis aber evtl.
> dennoch hilfreich.
> Nicht zuletzt wegen der kleinen aber feinen Anmerkung:
> Note: Use IP addresses instead of hostnames for Nginx
> :-)
> Letztlich wird aber sicherlich @netzarbeiter nochmals gegenlesen und
> final entscheiden.
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>

fkaminski commented:
Yepp, wie immer eine Gradwanderung :-)
Eine spätere Ergänzung/Verlinkung fände ich dann auch optimaler. 
Warten wir mal ab was @netzarbeiter dazu sagt.

netzarbeiter commented on docs/manual/guides/local-installation/devilbox.de.md:
```diff
@@ -86,34 +94,40 @@ deine IP-Adresse möglicherweise anders. Die IP-Adresse kann über den Befehl `d
 
-Du hast ein Verzeichnis (z. B. `contao4`) erstellt. Wechsle in dieses Verzeichnis und erstelle einen neuen 
-Unterordner `web`. Kopiere in diesen Ordner die Contao Manager `.phar` Datei und benenne die Datei um in `contao-manager.phar.php`. 
+Du hast ein Verzeichnis (z. B. `contao4`) erstellt. Wechsle in dieses Verzeichnis und erstelle einen neuen Unterordner `web` bzw. `public`.
```
```suggestion
Du hast ein Verzeichnis (z. B. `contao4`) erstellt. Wechsle in dieses Verzeichnis und erstelle einen neuen Unterordner `public` (bis und mit Contao 4.11 `web`).
```

netzarbeiter commented on docs/manual/guides/local-installation/devilbox.de.md:
```diff
@@ -86,34 +94,40 @@ deine IP-Adresse möglicherweise anders. Die IP-Adresse kann über den Befehl `d
-Du hast ein Verzeichnis (z. B. `contao4`) erstellt. Wechsle in dieses Verzeichnis und erstelle einen neuen 
-Unterordner `web`. Kopiere in diesen Ordner die Contao Manager `.phar` Datei und benenne die Datei um in `contao-manager.phar.php`. 
+Du hast ein Verzeichnis (z. B. `contao4`) erstellt. Wechsle in dieses Verzeichnis und erstelle einen neuen Unterordner `web` bzw. `public`.
+Kopiere in diesen Ordner die Contao Manager `.phar` Datei und benenne die Datei um in `contao-manager.phar.php`. 
```
```suggestion
Kopiere in diesen Ordner die Contao Manager `.phar` Datei und benenne die Datei in `contao-manager.phar.php` um. 
```

netzarbeiter commented on docs/manual/guides/local-installation/devilbox.de.md:
```diff
@@ -124,33 +138,41 @@ Die weitere Vorgehensweise ist dann identisch wie in [Contao installieren](../..
 Im Anschluss musst du die Devilbox neu starten. Im Devilbox Hauptverzeichnis liegen die Dateien `shell.sh` bzw. `shell.bat`. 
-Damit kannst du dich in den laufenden Devilbox PHP Container einklinken. Hier sind bereits [zahlreiche Tools](https://devilbox.readthedocs.io/en/latest/readings/available-tools.html) vorinstalliert. Auch `Composer`. Nach Aufruf befindest du dich im Container im 
-Verzeichnis `shared\http`. Zur Installation von z. B. Contao 4.8 in ein Verzeichnis `contao48` musst du lediglich eingeben:
+Damit kannst du dich in den laufenden Devilbox PHP Container (die `Devilbox-shell`) einklinken. Hier sind bereits [zahlreiche Tools](https://devilbox.readthedocs.io/en/latest/readings/available-tools.html) vorinstalliert. Auch `Composer`. Nach Aufruf befindest du dich im Container im
```
```suggestion
Damit kannst du dich in den laufenden Devilbox PHP-Container (die `Devilbox-shell`) einklinken. Hier sind bereits [zahlreiche Tools](https://devilbox.readthedocs.io/en/latest/readings/available-tools.html) vorinstalliert. Auch `Composer`. Nach Aufruf befindest du dich im Container im
```

netzarbeiter commented on docs/manual/guides/local-installation/devilbox.de.md:
```diff
@@ -124,33 +138,41 @@ Die weitere Vorgehensweise ist dann identisch wie in [Contao installieren](../..
 
-Im Anschluss kannst du den Container über `exit` verlassen und das Contao-Installtool aufrufen.
+{{% notice tip %}}
+Halte die Devilbox-Shell in einem separaten Ternminalfenster während deiner Arbeit offen. Contao-Kommandos, z.B. ein
```
```suggestion
Halte die Devilbox-Shell in einem separaten Ternminalfenster während deiner Arbeit offen. Contao-Kommandos, z. B. ein
```

netzarbeiter commented on docs/manual/guides/local-installation/devilbox.de.md:
````diff
@@ -124,33 +138,41 @@ Die weitere Vorgehensweise ist dann identisch wie in [Contao installieren](../..
+```
 
-## Angaben im Contao-Installtool
+geben in der Devilbox-Shell u.U. mehr Informationen preis als wenn sie unter dem Host ausgeführt werden.  
````
```suggestion
geben in der Devilbox-Shell u. U. mehr Informationen preis als wenn sie unter dem Host ausgeführt werden.  
```

netzarbeiter commented on docs/manual/guides/local-installation/devilbox.de.md:
```diff
@@ -165,3 +187,115 @@ Der Benutzer `root` mit leerem Passwort ist die Devilbox Standard Einstellung. D
+
+## Nützliche Informationen: Mehrere php Versionen parallel betreiben {{< ab devilbox version "3" >}} 
+
+Die Grundeinstellung nimmst du im devilbox Verzeichnis vor.
```
```suggestion
Die Grundeinstellung nimmst du im Devilbox-Verzeichnis vor.
```

netzarbeiter commented on docs/manual/guides/local-installation/devilbox.de.md:
```diff
@@ -165,3 +187,115 @@ Der Benutzer `root` mit leerem Passwort ist die Devilbox Standard Einstellung. D
+## Nützliche Informationen: Mehrere php Versionen parallel betreiben {{< ab devilbox version "3" >}} 
+
+Die Grundeinstellung nimmst du im devilbox Verzeichnis vor.
+Im Verzeichnis `compose` findest du die Datei `docker-compose.override.yml-php-multi.yml`. Kopiere die Datei in das devilbox Hautpverzeichnis und ändere den Dateinamen nach »docker-compose.override.yml«. 
```
```suggestion
Im Verzeichnis `compose` findest du die Datei `docker-compose.override.yml-php-multi.yml`. Kopiere die Datei in das Devilbox-Hautpverzeichnis und ändere den Dateinamen nach »docker-compose.override.yml«. 
```

netzarbeiter commented on docs/manual/guides/local-installation/devilbox.de.md:
```diff
@@ -165,3 +187,115 @@ Der Benutzer `root` mit leerem Passwort ist die Devilbox Standard Einstellung. D
+Die Grundeinstellung nimmst du im devilbox Verzeichnis vor.
+Im Verzeichnis `compose` findest du die Datei `docker-compose.override.yml-php-multi.yml`. Kopiere die Datei in das devilbox Hautpverzeichnis und ändere den Dateinamen nach »docker-compose.override.yml«. 
+
+Du hast ein Projekt <project>, das abweichend von der in der .env eingestellten php-Version (in unserem Beispiel 8.2) mit einer anderen php-Version - sagen wir 7.4 - laufen soll?
```
```suggestion
Du hast ein Projekt <project>, das abweichend von der in der .env eingestellten PHP-Version (in unserem Beispiel 8.2) mit einer anderen PHP-Version - sagen wir 7.4 - laufen soll?
```

netzarbeiter commented on docs/manual/guides/local-installation/devilbox.de.md:
```diff
@@ -165,3 +187,115 @@ Der Benutzer `root` mit leerem Passwort ist die Devilbox Standard Einstellung. D
+Im Verzeichnis `compose` findest du die Datei `docker-compose.override.yml-php-multi.yml`. Kopiere die Datei in das devilbox Hautpverzeichnis und ändere den Dateinamen nach »docker-compose.override.yml«. 
+
+Du hast ein Projekt <project>, das abweichend von der in der .env eingestellten php-Version (in unserem Beispiel 8.2) mit einer anderen php-Version - sagen wir 7.4 - laufen soll?
+Lege in diesem Projektverzeichnis ein Verzeichnis `.devilbox` und darin eine Datei `backend.cfg` an mit diesem Inhalt:
```
```suggestion
Lege in diesem Projektverzeichnis ein Verzeichnis `.devilbox` und darin eine Datei `backend.cfg` mit diesem Inhalt an:
```

netzarbeiter commented on docs/manual/guides/local-installation/devilbox.de.md:
````diff
@@ -165,3 +187,115 @@ Der Benutzer `root` mit leerem Passwort ist die Devilbox Standard Einstellung. D
+conf:phpfpm:tcp:php74:9000  
+```
+
+Starte devilbox wie folgt:
````
```suggestion
Starte Devilbox wie folgt:
```

netzarbeiter commented on docs/manual/guides/local-installation/devilbox.de.md:
```diff
@@ -165,3 +187,115 @@ Der Benutzer `root` mit leerem Passwort ist die Devilbox Standard Einstellung. D
+{{% /notice %}}
+
+{{% notice tip %}}
+Im devilbox Dashboard siehst du unter `Virtual Hosts` in der Spalte `Backend', welche php-Version tatsächlich für ein Projekt im Einsatz ist. (Noch gibt es dort einen kleinen Fehler: Auskommentierte Konfigurationen werden nicht als passiv erkannt.)
```
```suggestion
Im Devilbox-Dashboard siehst du unter `Virtual Hosts` in der Spalte `Backend', welche PHP-Version tatsächlich für ein Projekt im Einsatz ist. (Noch gibt es dort einen kleinen Fehler: Auskommentierte Konfigurationen werden nicht als passiv erkannt.)
```

netzarbeiter commented on docs/manual/guides/local-installation/devilbox.de.md:
```diff
@@ -86,34 +94,40 @@ deine IP-Adresse möglicherweise anders. Die IP-Adresse kann über den Befehl `d
 ## Contao-Installation vorbereiten
 
-Eine oder mehrere Contao-Installationen werden im Devilbox Verzeichnis **`data\www`** erstellt. Je Contao-Installation 
+Eine oder mehrere Contao-Installationen werden im Devilbox Verzeichnis **`data/www`** erstellt. Je Contao-Installation 
```
```suggestion
Eine oder mehrere Contao-Installationen werden im Devilbox-Verzeichnis **`data/www`** erstellt. Je Contao-Installation 
```

netzarbeiter commented on docs/manual/guides/local-installation/devilbox.de.md:
```diff
@@ -86,34 +94,40 @@ deine IP-Adresse möglicherweise anders. Die IP-Adresse kann über den Befehl `d
 ## Installation über den Contao Manager
 
-Starte `phpMyAdmin` im Devilbox Dashboard im Bereich `Tools\phpMyAdmin` und lege eine neue Datenbank an. Wechsle dann 
+Starte `phpMyAdmin` im Devilbox Dashboard im Bereich `Tools/phpMyAdmin` und lege eine neue Datenbank an. Wechsle dann 
```
```suggestion
Starte `phpMyAdmin` im Devilbox-Dashboard im Bereich `Tools/phpMyAdmin` und lege eine neue Datenbank an. Wechsle dann 
```

netzarbeiter commented on docs/manual/guides/local-installation/devilbox.de.md:
```diff
@@ -165,3 +187,115 @@ Der Benutzer `root` mit leerem Passwort ist die Devilbox Standard Einstellung. D
+
+Im [Devilbox-Handbuch](https://devilbox.readthedocs.io/en/latest/intermediate/configure-php-xdebug.html?highlight=xdebug#configure-php-xdebug-1) findest du eine nicht ganz aktuelle Dokumentation, die aber das grundsätzliche Vorgehen gut aufzeigt.
+
+Für Vscode funktioniert diese Konfiguration sehr gut:
```
```suggestion
In meiner IDE funktioniert folgende Konfiguration sehr gut:
```

Ernestopheles replied:
I am not sure if it then becomes clear that this configuration is for Vscode only.

netzarbeiter replied:
The configuration should not have anything to do with an IDE. We don't want to force people to use Vscode or any other IDE. They should be able to use Devilbox with their IDE.

Ernestopheles replied:
Also dann nehmen wir deinen Vorschlag. Wenn Nutzer*innen einer anderen IDE abweichende Konfis haben sollten, kann das ja hier noch ergänzt werden.

netzarbeiter commented on docs/manual/guides/local-installation/devilbox.de.md:
```diff
@@ -37,9 +37,9 @@ die Datei anschließend um in `.env`. In der neuen Datei kannst du jetzt deine K
 Die einzelnen Schritte (gerade auch für die Einträge `NEW_UID` und `NEW_GID`) sind in der [Devilbox Dokumentation](https://devilbox.readthedocs.io/en/latest/getting-started/install-the-devilbox.html#set-uid-and-gid) gut beschrieben. Für Contao selbst sollten die weiteren Einträge etwa so gesetzt werden:
 
-- `HTTPD_DOCROOT_DIR=web`
+- `HTTPD_DOCROOT_DIR=DOCUMENT-ROOT` (DOCUMENT-ROOT = `web` bzw. `public`, siehe auch [Hosting-Konfiguration](https://docs.contao.org/manual/de/installation/systemvoraussetzungen/#hosting-konfiguration))
```
```suggestion
- `HTTPD_DOCROOT_DIR=DOCUMENT-ROOT` (DOCUMENT-ROOT = `public` (bis und mit Contao 4.11 `web`), siehe auch [Hosting-Konfiguration](https://docs.contao.org/manual/de/installation/systemvoraussetzungen/#hosting-konfiguration))
```

netzarbeiter reviewed (changes requested):
Thank you, Ernst.

Can you please update the english version.

Ernestopheles committed:
Update docs/manual/guides/local-installation/devilbox.de.md

Co-authored-by: netzarbeiter <netzarbeiter@users.noreply.github.com>

Ernestopheles committed:
Update docs/manual/guides/local-installation/devilbox.de.md

Co-authored-by: netzarbeiter <netzarbeiter@users.noreply.github.com>

Ernestopheles committed:
Update docs/manual/guides/local-installation/devilbox.de.md

Co-authored-by: netzarbeiter <netzarbeiter@users.noreply.github.com>

Ernestopheles committed:
Update docs/manual/guides/local-installation/devilbox.de.md

Co-authored-by: netzarbeiter <netzarbeiter@users.noreply.github.com>

Ernestopheles committed:
Update docs/manual/guides/local-installation/devilbox.de.md

Co-authored-by: netzarbeiter <netzarbeiter@users.noreply.github.com>

Ernestopheles committed:
Update docs/manual/guides/local-installation/devilbox.de.md

Co-authored-by: netzarbeiter <netzarbeiter@users.noreply.github.com>

Ernestopheles committed:
Update docs/manual/guides/local-installation/devilbox.de.md

Co-authored-by: netzarbeiter <netzarbeiter@users.noreply.github.com>

Ernestopheles committed:
Update docs/manual/guides/local-installation/devilbox.de.md

Co-authored-by: netzarbeiter <netzarbeiter@users.noreply.github.com>

Ernestopheles committed:
Update docs/manual/guides/local-installation/devilbox.de.md

Co-authored-by: netzarbeiter <netzarbeiter@users.noreply.github.com>

Ernestopheles committed:
Update docs/manual/guides/local-installation/devilbox.de.md

Co-authored-by: netzarbeiter <netzarbeiter@users.noreply.github.com>

Ernestopheles commented:
> Thank you, Ernst.
> 
> Can you please update the english version.

pleasure, done

Ernestopheles commented:
> 
> Can you please update the english version.

done

Ernestopheles commented:
Going to do some corrections tomorrow (analogue to en)

Ernestopheles committed:
corrections for mistakes, being visible after Wysiwyg preview

Ernestopheles committed:
Merge branch 'contao:main' into patch-2

Ernestopheles committed:
Zwei PR zusammengefügt

Ernestopheles commented:
> Thank you, Ernst.
> 
> Can you please update the english version.

done

Ernestopheles committed:
Update docs/manual/guides/local-installation/devilbox.de.md

Co-authored-by: netzarbeiter <netzarbeiter@users.noreply.github.com>

Ernestopheles committed:
Update docs/manual/guides/local-installation/devilbox.de.md

Co-authored-by: netzarbeiter <netzarbeiter@users.noreply.github.com>

Ernestopheles committed:
Update docs/manual/guides/local-installation/devilbox.de.md

Co-authored-by: netzarbeiter <netzarbeiter@users.noreply.github.com>

Ernestopheles committed:
Update docs/manual/guides/local-installation/devilbox.de.md

Co-authored-by: netzarbeiter <netzarbeiter@users.noreply.github.com>

netzarbeiter committed:
Update devilbox.de.md

netzarbeiter committed:
Update devilbox.en.md

netzarbeiter reviewed (approved):
Thank you, Ernst
